src/options/generate/generator.py

```diff
@@ -320,3 +347,87 @@ def summarize_error(self, error):
         error_summary = conversation.chat(template_summarize_error.format(error=error))
         return error_summary
 
+    def refine_specification(self):
+        pm = get_random_employee('pm')
+        print(f'{pm.emoji}👋 Hi, I\'m {pm.name}, a PM at Jina AI. Gathering the requirements for our engineers.')
+        self.refine_task(pm)
+        self.refine_test(pm)
+        print(f'''
+{pm.emoji} 👍 Great, I will handover the following requirements to our engineers:
+Description of the microservice:
+{self.microservice_specification.task}
+Test scenario:
+{self.microservice_specification.test}
+''')
+
+    def refine_task(self, pm):
+        task_description = self.microservice_specification.task
+        if not task_description:
+            task_description = self.get_user_input(pm, 'What should your microservice do?')
+        messages = [
+            SystemMessage(content=system_task_introduction + system_task_iteration),
+        ]
+
+        while True:
+            conversation = self.gpt_session.get_conversation(messages, print_stream=os.environ['VERBOSE'].lower() == 'true', print_costs=False)
+            print('thinking...')
+            agent_response_raw = conversation.chat(task_description, role='user')
+
+            question = self.extract_content_from_result(agent_response_raw, 'prompt.txt', can_contain_code_block=False)
+            task_final = self.extract_content_from_result(agent_response_raw, 'task-final.txt', can_contain_code_block=False)
+            if task_final:
+                self.microservice_specification.task = task_final
+                break
+            if question:
+                task_description = self.get_user_input(pm, question)
+                messages.extend([HumanMessage(content=task_description)])
+            else:
+                task_description = self.get_user_input(pm, agent_response_raw + '\n: ')
+
+    def refine_test(self, pm):
```

Comment:
there is code duplication in `refine_task` and `refine_test`

Reply:
done